Core/Games/KerbalSpaceProgram2.cs

```diff
@@ -0,0 +1,204 @@
+        {
+            if (Platform.IsMac)
+            {
+                string installPath = Path.Combine(
+                    // This is "/Applications" in Mono on Mac
+                    Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles),
+                    "Kerbal Space Program 2"
+                );
+                return Directory.Exists(installPath) ? installPath : null;
+            }
+            return null;
+        }
+
+        public string PrimaryModDirectoryRelative => "SpaceWarp/Mods";
+
+        public string PrimaryModDirectory(GameInstance inst)
+            => CKANPathUtils.NormalizePath(
+                Path.Combine(inst.GameDir(), PrimaryModDirectoryRelative));
+
+        public string[] StockFolders => new string[]
+        {
+            "KSP2_x64_Data",
+            "MonoBleedingEdge",
+            "PDLauncher",
+        };
+
+        public string[] ReservedPaths => new string[]
+        {
+        };
+
+        public string[] CreateableDirs => new string[]
+        {
+            "SpaceWarp",
+            "SpaceWarp/Mods",
+            "BepInEx",
+            "BepInEx/plugins",
+        };
+
+        /// <summary>
+        /// Checks the path against a list of reserved game directories
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        public bool IsReservedDirectory(GameInstance inst, string path)
+            => path == inst.GameDir() || path == inst.CkanDir()
+                || path == PrimaryModDirectory(inst);
+
+        public bool AllowInstallationIn(string name, out string path)
+            => allowedFolders.TryGetValue(name, out path);
+
+        public void RebuildSubdirectories(GameInstance inst)
+        {
+        }
+
+        public string DefaultCommandLine =>
+                  Platform.IsUnix ? "./KSP2.x86_64 -single-instance"
+                : Platform.IsMac  ? "./KSP2.app/Contents/MacOS/KSP"
+                :                   "KSP2_x64.exe -single-instance";
+
+        public string[] AdjustCommandLine(string[] args, GameVersion installedVersion)
+            => args;
+
+        public List<GameVersion> KnownVersions
+            => new List<GameVersion>
+            {
+                new GameVersion(0, 1, 0, 0),
+            };
+
+        public GameVersion DetectVersion(DirectoryInfo where)
+            => GameVersion.Parse(
+                FileVersionInfo.GetVersionInfo(
+                    Path.Combine(where.FullName, "KSP2_x64.exe")).ProductVersion);
+
+        public string CompatibleVersionsFile => "compatible_game_versions.json";
+
+        public string[] BuildIDFiles => new string[]
+        {
+            "KSP2_x64.exe",
+        };
+
+        public Uri DefaultRepositoryURL => new Uri("https://github.com/KSP-CKAN/KSP2-CKAN-meta/archive/master.tar.gz");
+
+        public Uri RepositoryListURL => new Uri("https://raw.githubusercontent.com/KSP-CKAN/KSP2-CKAN-meta/master/repositories.json");
```

Comment:
Should we move our urls to a central constants location?

Reply:
No, I think these should stay here. They're only used for one purpose, and having to go hunt for them somewhere else would be extra hassle for no benefit.